### PR TITLE
chore(custom-resources): update lambda script to use certificate ARN as physical resource ID

### PR DESCRIPTION
**Pull request by Lou1415926**
Previously, the script use `service/<sorted aliases>` as the physical resource ID. This is found not to be the most accurate representation of the resource. 

A certificate is not uniquely defined by canonical domain, SANs and tags. Although the Copilot workflow always maintain only one such certificate, it should also be robust against unforeseen situation where multiple such certificates exist. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Review by iamhopaul123 — APPROVED** [review]

File: cf-custom-resources/lib/nlb-cert-validator-updater.js [hunk 1]
```diff
@@ -214,14 +209,23 @@ exports.handler = async function (event, context) {
...
                         .digest("hex")});
+                physicalResourceID = certificateARN; // Update the physical resource ID if a new certificate is created.
                 const options = await waitForValidationOptionsToBeReady(certificateARN, aliases);
                 await activate(options, certificateARN, loadBalancerDNS, loadBalancerHostedZoneID);
                 break;
             case "Delete":
-                let unusedOptions = await unusedValidationOptions(aliases, loadBalancerDNS);
-                await deleteCertificate(aliases);
+                if (!physicalResourceID.startsWith("arn:")) {
+                    // This means no certificate has been created, nor any records. Exit without doing anything.
```
Comment (efekarakus) [inline cf-custom-resources/lib/nlb-cert-validator-updater.js]: would this cause a weird exception if `physicalResourceID` is `undefined`?

File: cf-custom-resources/lib/nlb-cert-validator-updater.js [hunk 2]
```diff
@@ -192,13 +192,8 @@ exports.handler = async function (event, context) {
...
     };
 
-    // NOTE: If the aliases have changed, then we need to replace the certificate being used, as well as deleting/adding
-    // validation records and A records. In general, any change in aliases indicate a "replacement" of the resources
-    // managed by the custom resource lambda; on the contrary, the same set of aliases indicate that there is no need to
-    // replace or update the certificate, nor the validation records or A records. Hence, we can use this as the physicalResourceID.
     let aliasesSorted = [...aliases].sort().join(",");
-    const physicalResourceID = `/${serviceName}/${aliasesSorted}`;
-
+    let physicalResourceID = event.PhysicalResourceId; // Keep old physical resource ID unchanged by default.
```
Comment (efekarakus) [inline cf-custom-resources/lib/nlb-cert-validator-updater.js]: nit:
```suggestion
    let physicalResourceID = event.PhysicalResourceId; // The certificate ARN. By default, keep old physical resource ID unchanged.
```

**Review by efekarakus — APPROVED** [review]
Looks good, just a question. Feel free to remove label once addressed!

File: cf-custom-resources/lib/nlb-cert-validator-updater.js (hunk 1 above)
Comment (Lou1415926) [reply cf-custom-resources/lib/nlb-cert-validator-updater.js]: From what I have observed, the `physicalResourceID` would be provided as a weird looking random string (e.g. "a3dkdfanrdsa3") to the "Delete" event when the creation fails. However I can't find the documentation for this behavior, so I added an extra logic that says: `if (!physicalResourceID || !physicalResourceID.startsWith("arn:"))` to deal with the possibility when the physical resource ID is undefined.

**Pull request merged**